Can not update SHA1 within a module

I'm trying to upgrade some libs in `plugins/discovery-azure`. So I need to update the `SHA1` files.

```sh
cd plugins/discovery-azure
gradle updateSHAs
```

It gives:

```
...
=======================================
Elasticsearch Build Hamster says Hello!
=======================================
  Gradle Version        : 2.8
  OS Info               : Mac OS X 10.11.1 (x86_64)
  JDK Version           : Oracle Corporation 1.8.0_60 [Java HotSpot(TM) 64-Bit Server VM 25.60-b23]
:plugins:discovery-azure:updateShas FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':plugins:discovery-azure:updateShas'.
> Could not find property 'SHA_EXTENSION' on task ':plugins:discovery-azure:updateShas'.
```

Strange. I tried from the root project dir and it failed with:

```
:core:updateShas FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core:updateShas'.
> java.io.FileNotFoundException: /Users/dpilato/Documents/Elasticsearch/dev/es-gradle/elasticsearch/core/licenses
```

@rjernst Any idea?

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/DependencyLicensesTask.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/DependencyLicensesTask.groovy
@@ -69,7 +69,7 @@ import java.util.regex.Pattern
  * </pre>
  */
 public class DependencyLicensesTask extends DefaultTask {
-    private static final String SHA_EXTENSION = '.sha1'
+    static final String SHA_EXTENSION = '.sha1'
 
     // TODO: we should be able to default this to eg compile deps, but we need to move the licenses
     // check from distribution to core (ie this should only be run on java projects)

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/UpdateShasTask.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/UpdateShasTask.groovy
@@ -35,6 +35,7 @@ public class UpdateShasTask extends DefaultTask {
 
     public UpdateShasTask() {
         description = 'Updates the sha files for the dependencyLicenses check'
+        onlyIf { parentTask.licensesDir.exists() }
     }
 
     @TaskAction
@@ -42,13 +43,13 @@ public class UpdateShasTask extends DefaultTask {
         Set<File> shaFiles = new HashSet<File>()
         parentTask.licensesDir.eachFile {
             String name = it.getName()
-            if (name.endsWith(SHA_EXTENSION)) {
+            if (name.endsWith(DependencyLicensesTask.SHA_EXTENSION)) {
                 shaFiles.add(it)
             }
         }
         for (File dependency : parentTask.dependencies) {
             String jarName = dependency.getName()
-            File shaFile = new File(parentTask.licensesDir, jarName + SHA_EXTENSION)
+            File shaFile = new File(parentTask.licensesDir, jarName + DependencyLicensesTask.SHA_EXTENSION)
             if (shaFile.exists() == false) {
                 logger.lifecycle("Adding sha for ${jarName}")
                 String sha = MessageDigest.getInstance("SHA-1").digest(dependency.getBytes()).encodeHex().toString()

